packages/@aws-cdk/cloudformation-include/test/test-templates/fn-sub-valid-map.json

```diff
@@ -0,0 +1,17 @@
+{
+  "Resources": {
+    "Bucket": {
+      "Type": "AWS::S3::Bucket",
+      "Properties": {
+        "BucketName": { "Fn::Sub": [ "${Region}-foo-${!Immediate}-foo-${AnotherBucket}-${AnotherBucket.DomainName}-${Name}", 
+          { "Name": { "Ref" : "AnotherBucket" }, "Region": { "Fn::Base64": "AWS::Region" }} ]}
+      }
+    },
+    "AnotherBucket": {
+      "Type": "AWS::S3::Bucket",
+      "Properties": {
+        "BucketName": "another-bucket"
+      }
+    }
+  }
+}
```

Comment:
I feel like this test is repeating the same things as the string test (escaping, simple references, etc.), and I'm not sure what's the point of that.